library/core/src/iter/traits/iterator.rs

```diff
@@ -719,24 +717,17 @@ pub trait Iterator {
 
     /// Creates an iterator that both filters and maps.
     ///
-    /// The closure must return an [`Option<T>`]. `filter_map` creates an
-    /// iterator which calls this closure on each element. If the closure
-    /// returns [`Some(element)`][`Some`], then that element is returned. If the
-    /// closure returns [`None`], it will try again, and call the closure on the
-    /// next element, seeing if it will return [`Some`].
+    /// Given an element the closure must return `Some(value)` or `None`. The
+    /// returned iterator will only yield a value when the closure returns
+    /// `Some`. The values inside the `Some`s are yielded.
```

Comment:
This seems a little verbose, but I'm not sure how better to word it. It seems like it's saying the same thing 3 times sort of.

Reply:
I spent quite some time trying to get these three sentences down to 2 (or even better one), but failed.

It's hard to explain concisely that it is not the element that is returned, but the innards of the `Some` that is returned.

I'm open to suggestions, but also I'd like to move forward. You can always tweak this in a later PR.

Reply:
Would something like this be any better?

> The returned iterator yields only the `value`s for which the supplied closure returns `Some(value)`.

Reply:
Ooh, I like that!